Userland/Applications/PixelPaint/Image.cpp

```diff
@@ -678,7 +678,8 @@ ErrorOr<void> Image::resize(Gfx::IntSize new_size, Gfx::Painter::ScalingMode sca
             selected_layer_index = i;
 
         Gfx::IntPoint new_location(scale_x * new_layer->location().x(), scale_y * new_layer->location().y());
-        TRY(new_layer->resize(new_size, new_location, scaling_mode, Layer::NotifyClients::No));
+        Gfx::IntSize new_layer_size(scale_x * new_layer->size().width(), scale_y * new_layer->size().height());
+        TRY(new_layer->resize(new_layer_size, new_location, scaling_mode, Layer::NotifyClients::No));
```

Comment:
astonishing change; i would be guessing that computing the size is responsibility of a caller. or that itʼll be done when calling `resize` on the layer… this looks like weʼre covering a middle ground here. 🤔

i feel this might need a revisit. (care to check if it deserves a `TODO` or `FIXME`? 🙇🏻‍♂️)

it seems that `resize` is somewhat a misnomer if it deals with size _and_ location… maybe something more descriptive like `move_and_resize`, `apply`, or `set`? (not sure what exactly is done inside of that function—that might be a key for the proper name.)

Reply:
I've updated the PR to use a `Layer::resize()` method that you pass a new bounding rect for the layer, rather than passing a size and location individually. This is how the scaling methods in `Gfx::Painter` work and it seems clearer than what was there before.

I have also removed the other overloads of `Layer::resize()`, as they weren't used anywhere else and they seem like an easy way to introduce bugs like the one this PR fixes.